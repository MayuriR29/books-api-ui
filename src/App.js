import React, { Component } from "react";
import "./App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }
  render() {
    return (
      <div>
        {this.state.books.map(book => {
          return <li key={book._id}>{book.title}</li>;
        })}
      </div>
    );
  }
  async componentDidMount() {
    const url =
      process.env.REACT_APP_BOOKS_API || "http://localhost:3000/books";
    const response = await fetch(url);
    console.log("response", response);
    const books = await response.json();
    console.log("books", books);
    this.setState({
      books: books
    });
  }
}

export default App;
