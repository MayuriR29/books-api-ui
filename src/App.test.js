import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders a div with some snapshot", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const output = renderer.getRenderOutput();
  // expect(output.type).toBe("div"); not required
  expect(output).toMatchSnapshot();
});
