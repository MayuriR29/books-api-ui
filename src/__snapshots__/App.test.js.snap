// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a div with some snapshot 1`] = `
<div>
  <Books />
  <Authors />
</div>
`;
